packages/core/src/config/config.ts

```diff
@@ -712,9 +749,14 @@ export function transformConfig(rawConfig: DeprecatedRawConfig | RawConfig): Raw
       )} instead. Read more about this change: https://redocly.com/docs/api-registry/guides/migration-guide-config-file/#changed-properties\n`,
     );
   }
+  
   return {
     'features.openapi': referenceDocs,
     apis: transformApiDefinitionsToApis(apiDefinitions),
+    lint: lint && {
```

Comment:
I'm afraid it won't work correctly with nested configs we've just implemented (see [here](https://github.com/Redocly/openapi-cli/pull/616)). It looks like `transformConfig` will fire each time a nested config is loaded and thus convert all `assertions/...` entries into the single `assertions` rule. And then, it may be overwritten by the following `assertions` arrays from other configs if such exist. 
@adamaltman, is the actual intent to inherit assertion rules from the `extends` and then merge with the root rules, isn't it?

Reply:
@tatomyr yes. Of course, if there is a conflict on the individual assert name then it would override. For example,

**redocly.yaml**
```yaml
lint:
  extends:
    ./another.yaml
  rules:
    assert/operation-get:
      context:
      - type: Operation
        matchParentKeys: [get]
      subject: Operation
      property: operationId
      message: Operation id for get operation should be camelCase
      casing: camelCase
```

**another.yaml**
```yaml
lint:
  extends:
    ./another.yaml
  rules:
    assert/operation-get:
      context:
      - type: Operation
        matchParentKeys: [get]
      subject: Operation
      property: operationId
      message: Operation id for get operation should be PascalCase
      casing: PascalCase
    assert/operation-id-camel-case:
      subject: Operation
      property: operationId
      message: Operation id should be camelCase
      casing: camelCase
```

I would expect the previous two files to actually merge this way:

```yaml
lint:
  rules:
    assert/operation-get:
      context:
      - type: Operation
        matchParentKeys: [get]
      subject: Operation
      property: operationId
      message: Operation id for get operation should be camelCase
      casing: camelCase
    assert/operation-id-camel-case:
      subject: Operation
      property: operationId
      message: Operation id should be camelCase
      casing: camelCase

```

Because priority is given to the `redocly.yaml` file and there is one conflict where the priority matters.

Thanks for spotting this issue @tatomyr

Reply:
This is a good catch. I'll handle that once the [nested config PR](https://github.com/Redocly/openapi-cli/pull/616) is merged.

Reply:
Thanks @volodymyr-rutskyi

Reply:
@tatomyr The PR is now updated to handle nested configs and is ready to review again: https://github.com/Redocly/openapi-cli/pull/662/commits/69729babe4a71160ba67315a7971047d0935c3dd